win32ss/gdi/ntgdi/print.c

```diff
@@ -220,14 +233,24 @@ NtGdiExtEscape(
 freeout:
          if ( SafeInData )
             ExFreePoolWithTag ( SafeInData, GDITAG_TEMP );
-         DC_UnlockDc(pDC);
+         PDEVOBJ_vRelease(ppdev);
+         SURFACE_ShareUnlockSurface(psurf);
```

Comment:
This cleanup is getting long. Would some sort of `goto Exit` make sense?

Reply:
Makes sense.